src/gui/SetupTotpDialog.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (C) 2017 Weslly Honorato <﻿weslly@protonmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "SetupTotpDialog.h"
+#include "ui_SetupTotpDialog.h"
+#include "totp/totp.h"
+
+
+SetupTotpDialog::SetupTotpDialog(DatabaseWidget* parent, Entry* entry)
+    : QDialog(parent)
+    , m_ui(new Ui::SetupTotpDialog())
+{
+    m_entry = entry;
+    m_parent = parent;
+
+    m_ui->setupUi(this);
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    this->setFixedSize(this->sizeHint());
+
+    connect(m_ui->buttonBox, SIGNAL(rejected()), SLOT(close()));
+    connect(m_ui->buttonBox, SIGNAL(accepted()), SLOT(setupTotp()));
+    connect(m_ui->customSettingsCheckBox, SIGNAL(toggled(bool)), SLOT(toggleCustom(bool)));
+}
+
+
+void SetupTotpDialog::setupTotp()
+{
+    quint8 digits;
+
+    if (m_ui->radio8Digits->isChecked()) {
+        digits = 8;
+    } else {
+        digits = 6;
+    }
```

Comment:
@louib That would assume the default was always 6 but it wouldn't work if the default was 8 and the user checked 6 instead. Realistically the default is always 6 but that could cause problems if someone compiled their own version with 8 as the default.